File: frontend/src/app/order/order.ts
import {Customer} from '../customer/customer';
import {Orderline} from './orderline';

export class Order {
  id: number;
  customerId: number;
  status: number;
  discount: number;
  shippingCost: number;
  subTotal: number;
  vatTotal: number;
  total: number;

  customer: Customer;
  orderlines: Orderline[];
}

File: pdf/src/main/java/be/bakmix/eindproject/pdf/service/dto/Order.java
package be.bakmix.eindproject.pdf.service.dto;

import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

/**
 * Order DTO with Customer and Orderline
 */
@Data
public class Order {
    private Long id;
    private Long customerId;
    private LocalDate date;
    private Long status;
    private BigDecimal discount;
    private  BigDecimal shippingCost;
    private BigDecimal subTotal;
    private BigDecimal vatTotal;
    private BigDecimal total;

    private Customer customer;
    private List<Orderline> orderlines = new ArrayList<>();
}
